src/sage/combinat/permutation.py

```diff
@@ -758,13 +843,42 @@ def __next__(self):
             sage: p = Permutation([4,3,2,1])
             sage: next(p)
             False
+            sage: p = Permutation([1, 2, 3], algorithm='sjt')
+            sage: p = next(p); p
+            [1, 3, 2]
+            sage: p = next(p); p
+            [3, 1, 2]
 
         TESTS::
 
             sage: p = Permutation([])
             sage: next(p)
             False
+            sage: p = Permutation([], algorithm='sjt')
+            sage: next(p)
+            False
+            sage: p = Permutation([1], algorithm='sjt')
+            sage: next(p)
+            False
+            sage: l = [1, 2, 3, 4]
+            sage: s = set()
+            sage: p = Permutation(l, algorithm='sjt')
+            sage: for _ in range(factorial(len(l))):
+            ....:     s.add(p)
+            ....:     p = p.next()
+            ....:
+            sage: p
+            False
+            sage: assert(len(s)) == factorial(len(l))
         """
+        if self.__sjt is not None:
+            # Ensure the same permutation is yielded when called multiple times
+            # without reassigning
+            p, directions = self.__sjt.next()
+            if p is False:
+                return False
+            return Permutations()(p, algorithm='sjt', directions=directions)
```

Comment:
Seems to work for me:
```suggestion
            return Permutation(p, algorithm='sjt', directions=directions)
```

Reply:
Tbh, I don't know the double `()()` syntax on calls in python so I just "copied" code there.
Are we sure this isn't changing something we don't know about ?

Reply:
It's not double `()()` syntax. `Permutations()` is creating a [Parent](https://doc.sagemath.org/html/en/reference/structure/sage/structure/parent.html#sage.structure.parent.Parent) of all permutations, and the `__call__` method on that tries to further convert the argument to the `Element` class of that parent class (afaik). So I think `Permutation` should be faster, but I'm not sure if anything breaks.

Leave it if you want, or try it and see any CI breaks :p

Reply:
I confirm it breaks the CI on my end.